filestore/dashboards.go

```diff
@@ -0,0 +1,209 @@
+package filestore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/influxdata/chronograf"
+)
+
+// DashExt is the the file extension searched for in the directory for dashboard files
+const DashExt = ".dashboard"
+
+var _ chronograf.DashboardsStore = &Dashboards{}
+
+// Dashboards are JSON dashboards stored in the filesystem
+type Dashboards struct {
+	Dir     string                                      // Dir is the directory containing the dashboards.
+	Load    func(string, interface{}) error             // Load loads string name and dashbaord passed in as interface
+	Create  func(string, interface{}) error             // Create will write dashboard to file.
+	ReadDir func(dirname string) ([]os.FileInfo, error) // ReadDir reads the directory named by dirname and returns a list of directory entries sorted by filename.
+	Remove  func(name string) error                     // Remove file
+	IDs     chronograf.ID                               // IDs generate unique ids for new dashboards
+	Logger  chronograf.Logger
+}
+
+// NewDashboards constructs a dashboard store wrapping a file system directory
+func NewDashboards(dir string, ids chronograf.ID, logger chronograf.Logger) chronograf.DashboardsStore {
+	return &Dashboards{
+		Dir:     dir,
+		Load:    load,
+		Create:  create,
+		ReadDir: ioutil.ReadDir,
+		Remove:  os.Remove,
+		IDs:     ids,
+		Logger:  logger,
+	}
+}
+
+func dashboardFile(dir string, dashboard chronograf.Dashboard) string {
+	base := fmt.Sprintf("%s%s", dashboard.Name, DashExt)
+	return path.Join(dir, base)
+}
+
+func load(name string, resource interface{}) error {
+	octets, err := ioutil.ReadFile(name)
+	if err != nil {
+		return fmt.Errorf("resource %s not found", name)
+	}
+	return json.Unmarshal(octets, resource)
+}
+
+func create(file string, resource interface{}) error {
+	h, err := os.Create(file)
+	if err != nil {
+		return err
+	}
+	defer h.Close()
+
+	octets, err := json.MarshalIndent(resource, "    ", "    ")
```

Comment:
would it be safer / possible to put these indentations into a string constant?

Reply:
I don't think it need to be moved into a global variable.  This is just to make the output JSON look better for a person to consume.  

Adding a global variable or constant would disrupt "locality" of code in this case and decrease readability.